controllers/controllers.errors.js

```diff
@@ -9,13 +9,14 @@ exports.handleCustomErrors =(err, req, res, next) => {
   }
 }
 exports.handlePsqlErrors = (err, req, res, next) => {
-  if (err.code === "22P02") {
+  if (err.code === "22P02" || err.code === "23502") {
     res.status(400).send({ msg: "Bad request" });
-  } else {
-    next(err);
-  }
+  } else if (err.code === "23503") {
+    res.status(404).send({ msg: "Not found" });
+  } else next(err);
```

Comment:
Spot on! Very nicely done to let psql do the hard work for you on these errors and then check for the codes 👍